scripts/test/Muon/results_tab_widget/results_tab_model_test.py

```diff
@@ -149,222 +249,166 @@ def side_effect(name):
-    """
-    Create a list of fits
-    :param input_workspaces: The input workspaces
-    :param function_name: The name of the function
-    :param parameters: The parameters list
-    :return: A list of Fits
-    """
-    fits = []
-    for name in input_workspaces:
-        parameter_workspace = mock.NonCallableMagicMock()
-        parameter_workspace.workspace.toDict.return_value = parameters
-        parameter_workspace.workspace_name = name + '_Parameters'
-        fits.append(FitInformation(parameter_workspace, function_name, name))
+    def test_create_results_table_with_mixed_global_non_global_raises_error(
+            self):
+        parameters = OrderedDict([('f0.Height', (100, 0.1)),
+                                  ('f1.Height', (90, 0.001)),
+                                  ('Cost function value', (1.5, 0))])
+        fits_func1= create_test_fits(('ws1', ), 'func1', parameters)
+        fits_globals = create_test_fits(('ws2', ), 'func1', parameters,
+                                        global_parameters=['Height'])
+        model = ResultsTabModel(FittingContext(fits_func1 + fits_globals))
 
-    return fits
+        selected_results = [('ws1', 0), ('ws2', 1)]
+        self.assertRaises(RuntimeError, model.create_results_table, [],
+                          selected_results)
 
+    # ---------------------- Private helper functions -------------------------
 
-def create_test_fits_with_logs(input_workspaces, function_name, parameters,
-                               logs):
-    """
-    Create a list of fits with time series logs on the workspaces
-    :param input_workspaces: See create_test_fits
-    :param function_name: See create_test_fits
-    :param parameters: See create_test_fits
-    :param logs: A list of log names to create
-    :return: A list of Fits with workspaces/logs attached
-    """
-    fits = create_test_fits(input_workspaces, function_name, parameters)
-    for fit, workspace_name in zip(fits, input_workspaces):
-        test_ws = create_test_workspace(workspace_name)
-        run = test_ws.run()
-        # populate with log data
-        for index, name in enumerate(logs):
-            tsp = FloatTimeSeriesProperty(name)
-            tsp.addValue("2019-05-30T09:00:00", float(index))
-            tsp.addValue("2019-05-30T09:00:05", float(index + 1))
-            run.addProperty(name, tsp, replace=True)
-
-        fit.input_workspace = workspace_name
+    def _assert_table_matches_expected(self, expected_cols, expected_content,
+                                       table, table_name):
+        self.assertTrue(isinstance(table, ITableWorkspace))
+        self.assertTrue(table_name in AnalysisDataService.Instance())
+        self.assertEqual(len(expected_cols), table.columnCount())
+        self.assertEqual(len(expected_content), table.rowCount())
+        self.assertEqual(expected_cols, table.getColumnNames())
 
-    return fits
+        print("expected=", end="")
```

Comment:
do we want these print statements?

Reply:
Nice catch. They were from some debugging that I thought I deleted.